systeminit/si
# Fix func argument type switching and refactor change set test helpers

## Description

This PR fixes updating func argument types for attribute funcs by using the correct content hash when creating new func argument nodes. Two new tests in `save_func::attribute` and `func::argument` test the functionality.

As part of this change, the `SavedFunc` type falls out and `SaveFuncResponse` returns. `Func` retrievals are streamlined through `FuncView::assemble`.

This PR also provides major quality of life fixes when test authoring. All functions and helpers related to change sets are co-located in a new module and driven by a new unit struct. The `test_harness` module has been removed and replaced with `expand_helpers`. Remaining functions were moved to the `helpers` module.

<img src="https://media1.giphy.com/media/YIXSUxRLwe5wyPlpeX/giphy.gif"/>

## Test Changes

- Replace all manual usage of commits with the new helpers
- Ensure all commits are checked for conflicts
- Remove all `old-engine-testing-layout` files
- Move all attribute save func tests to a sub-module

## Misc Changes

- Rename `view` to `bags` and change the corresponding struct name for `FuncAssociation` two-way bags
- Move the `summary` module out from underneath the `func::view` module since it is a wholly separate domain and add a new error and result type
- Optimize existing func argument and relevant code with newer patterns when using the new engine
- Move `FuncArgumentBag` to the new `bag` module
- Remove dead code in integration tests and in misc locations
